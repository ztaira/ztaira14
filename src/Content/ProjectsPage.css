.ProjectContent {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin: 80px 4% 50px 4%;
}

.ProjectList {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  flex-wrap: wrap;
  width: 100%;
}
