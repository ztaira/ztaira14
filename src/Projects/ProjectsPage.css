.ProjectContent {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin: 80px 2% 50px 2%;
  padding: 20px 10px;
  background-color: #fafafa;
  border-radius: 7px;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
}

.ProjectList {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  flex-wrap: wrap;
  width: 100%;
}

.ProjectFilterToggles {
  display: flex;
  flex-direction: row;
  justify-content: center;
}
