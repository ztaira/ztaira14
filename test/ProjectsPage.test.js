import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ProjectsPage from '../src/Content/ProjectsPage.jsx';
import ProjectsBackup from '../src/Content/ProjectsBackup.js';
// import ReactTestUtils from 'react-dom/test-utils';

beforeEach(function() {
  this.projectspage = ReactDOM.render(
    <ProjectsPage />,
    document.body.appendChild(document.createElement('div'))
  );
});

it('has not changed since the last snapshot', function() {
  const projectspage = renderer.create(
    <ProjectsPage />
  );
  let projectspage_json = projectspage.toJSON();
  expect(projectspage_json).toMatchSnapshot();
});

it('contains the expected sort buttons', function() {
  expect(this.projectspage.state.sortButtons).toEqual(
    [
      {
        'label': 'A-Z',
        'updateFunction': this.projectspage.UpdateDisplayedProjectsOnSortButtonClick,
        'state': false,
        'updateArgument': this.projectspage.SortFunctionAlphabeticalByName,
      },
      {
        'label': 'Z-A',
        'updateFunction': this.projectspage.UpdateDisplayedProjectsOnSortButtonClick,
        'state': false,
        'updateArgument': this.projectspage.SortFunctionReverseAlphabeticalByName,
      },
      {
        'label': 'Newest',
        'updateFunction': this.projectspage.UpdateDisplayedProjectsOnSortButtonClick,
        'state': false,
        'updateArgument': this.projectspage.SortFunctionNewestDateFirst,
      },
      {
        'label': 'Oldest',
        'updateFunction': this.projectspage.UpdateDisplayedProjectsOnSortButtonClick,
        'state': false,
        'updateArgument': this.projectspage.SortFunctionOldestDateFirst,
      },
    ]);
});

it('contains the expected filter buttons', function() {
  expect(this.projectspage.state.filterButtons).toEqual(
  [
    {
      'label': 'All',
      'updateFunction': this.projectspage.UpdateDisplayedProjectsOnFilterButtonClick,
      'state': false,
      'updateArgument': this.projectspage.RemoveNonSourceRepos,
    },
    {
      'label': 'Python',
      'updateFunction': this.projectspage.UpdateDisplayedProjectsOnFilterButtonClick,
      'state': false,
      'updateArgument': this.projectspage.RemoveNonPythonRepos,
    },
    {
      'label': 'JavaScript',
      'updateFunction': this.projectspage.UpdateDisplayedProjectsOnFilterButtonClick,
      'state': false,
      'updateArgument': this.projectspage.RemoveNonJavascriptRepos,
    },
    {
      'label': 'C++',
      'updateFunction': this.projectspage.UpdateDisplayedProjectsOnFilterButtonClick,
      'state': false,
      'updateArgument': this.projectspage.RemoveNonCRepos,
    },
    {
      'label': 'Misc',
      'updateFunction': this.projectspage.UpdateDisplayedProjectsOnFilterButtonClick,
      'state': false,
      'updateArgument': this.projectspage.RemovePythonJavascriptCRepos,
    },
    {
      'label': 'Hack-A-Week',
      'updateFunction': this.projectspage.UpdateDisplayedProjectsOnFilterButtonClick,
      'state': false,
      'updateArgument': this.projectspage.RemoveNonHackAWeekRepos,
    },
  ]);
});

it('has a working filter function to remove non source repos', function() {
  let project = { fork: false };
  expect(this.projectspage.RemoveNonSourceRepos(project)).toBe(true);
  project.fork = true;
  expect(this.projectspage.RemoveNonSourceRepos(project)).toBe(false);
});

it('has a working filter function to remove non C repos', function() {
  let project = { language: 'C++' };
  expect(this.projectspage.RemoveNonCRepos(project)).toBe(true);
  project.language = 'C';
  expect(this.projectspage.RemoveNonCRepos(project)).toBe(true);
  project.language = 'Python';
  expect(this.projectspage.RemoveNonCRepos(project)).toBe(false);
});

it('has a working filter function to remove non Python repos', function() {
  let project = { language: 'Python' };
  expect(this.projectspage.RemoveNonPythonRepos(project)).toBe(true);
  project.language = 'Javascript';
  expect(this.projectspage.RemoveNonPythonRepos(project)).toBe(false);
});

it('has a working filter function to remove non Javascript repos', function() {
  let project = { language: 'JavaScript' };
  expect(this.projectspage.RemoveNonJavascriptRepos(project)).toBe(true);
  project.language = 'Python';
  expect(this.projectspage.RemoveNonJavascriptRepos(project)).toBe(false);
});

it('has a working filter function to remove Python/JS/C repos', function() {
  let project = { language: 'Python' };
  expect(this.projectspage.RemovePythonJavascriptCRepos(project)).toBe(false);
  project.language = 'JavaScript';
  expect(this.projectspage.RemovePythonJavascriptCRepos(project)).toBe(false);
  project.language = 'C';
  expect(this.projectspage.RemovePythonJavascriptCRepos(project)).toBe(false);
  project.language = 'C++';
  expect(this.projectspage.RemovePythonJavascriptCRepos(project)).toBe(false);
  project.language = '';
  expect(this.projectspage.RemovePythonJavascriptCRepos(project)).toBe(true);
});

it('has a working filter function to remove non Hack-A-Week repos', function() {
  let project = { description: 'Hack-A-Week thing' };
  expect(this.projectspage.RemoveNonHackAWeekRepos(project)).toBe(true);
  project.description = 'Hello world';
  expect(this.projectspage.RemoveNonHackAWeekRepos(project)).toBe(false);
});

it('has a working alphabetical sort function', function() {
  let a = { name: 'a' };
  let b = { name: 'b' };
  expect(this.projectspage.SortFunctionAlphabeticalByName(a, b))
    .toBe(-1);
  a.name = 'c';
  expect(this.projectspage.SortFunctionAlphabeticalByName(a, b))
    .toBe(1);
  b.name = 'c';
  expect(this.projectspage.SortFunctionAlphabeticalByName(a, b))
    .toBe(0);
  a.name = '1';
  expect(this.projectspage.SortFunctionAlphabeticalByName(a, b))
    .toBe(-1);
});

it('has a working reverse alphabetical sort function', function() {
  let a = { name: 'a' };
  let b = { name: 'b' };
  expect(this.projectspage.SortFunctionReverseAlphabeticalByName(a, b))
    .toBe(1);
  a.name = 'c';
  expect(this.projectspage.SortFunctionReverseAlphabeticalByName(a, b))
    .toBe(-1);
  b.name = 'c';
  expect(this.projectspage.SortFunctionReverseAlphabeticalByName(a, b))
    .toBe(0);
  a.name = '1';
  expect(this.projectspage.SortFunctionReverseAlphabeticalByName(a, b))
    .toBe(1);
});

it('has a working newest-first sort function', function() {
  let a = { created_at: '2017-03-19T13:29:14Z'};
  let b = { created_at: '2016-04-10T04:20:26Z'};
  expect(this.projectspage.SortFunctionNewestDateFirst(a, b)).toBe(-1);
  b.created_at = '2017-03-19T13:29:14Z';
  a.created_at = '2016-04-10T04:20:26Z';
  expect(this.projectspage.SortFunctionNewestDateFirst(a, b)).toBe(1);
  b.created_at = '2016-04-10T04:20:26Z';
  expect(this.projectspage.SortFunctionNewestDateFirst(a, b)).toBe(0);
});

it('has a working oldest-first sort function', function() {
  let a = { created_at: '2017-03-19T13:29:14Z'};
  let b = { created_at: '2016-04-10T04:20:26Z'};
  expect(this.projectspage.SortFunctionOldestDateFirst(a, b)).toBe(1);
  b.created_at = '2017-03-19T13:29:14Z';
  a.created_at = '2016-04-10T04:20:26Z';
  expect(this.projectspage.SortFunctionOldestDateFirst(a, b)).toBe(-1);
  b.created_at = '2016-04-10T04:20:26Z';
  expect(this.projectspage.SortFunctionOldestDateFirst(a, b)).toBe(0);
});
