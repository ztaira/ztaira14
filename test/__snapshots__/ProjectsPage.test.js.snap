exports[`test has not changed since the last snapshot 1`] = `
<div
  className="ProjectContent"
  role="main">
  <div
    className="ProjectFilterToggles">
    <div>
      <button
        aria-haspopup={true}
        id="Filter"
        onClick={[Function]}>
        Filter by: All
      </button>
      <div
        className="ModalBackground"
        style={
          Object {
            "display": "none",
            "height": 768,
            "width": 1024,
          }
        }>
        <div
          className="ProjectFilterBoundingBox">
          <div
            className="ProjectFilters">
            <button
              className="ProjectFilter"
              id="All"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#666666",
                  "borderColor": "#666666",
                  "color": "#ffffff",
                }
              }>
              All
            </button>
            <button
              className="ProjectFilter"
              id="Python"
              onClick={[Function]}
              style={Object {}}>
              Python
            </button>
            <button
              className="ProjectFilter"
              id="JavaScript"
              onClick={[Function]}
              style={Object {}}>
              JavaScript
            </button>
            <button
              className="ProjectFilter"
              id="C++"
              onClick={[Function]}
              style={Object {}}>
              C++
            </button>
            <button
              className="ProjectFilter"
              id="Misc"
              onClick={[Function]}
              style={Object {}}>
              Misc
            </button>
            <button
              className="ProjectFilter"
              id="Hack-A-Week"
              onClick={[Function]}
              style={Object {}}>
              Hack-A-Week
            </button>
            <button
              className="ProjectFilter"
              id="Starred"
              onClick={[Function]}
              style={Object {}}>
              Starred
            </button>
          </div>
        </div>
      </div>
    </div>
    <div>
      <button
        aria-haspopup={true}
        id="Sort"
        onClick={[Function]}>
        Sort by: Newest
      </button>
      <div
        className="ModalBackground"
        style={
          Object {
            "display": "none",
            "height": 768,
            "width": 1024,
          }
        }>
        <div
          className="ProjectFilterBoundingBox">
          <div
            className="ProjectFilters">
            <button
              className="ProjectFilter"
              id="Newest"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#666666",
                  "borderColor": "#666666",
                  "color": "#ffffff",
                }
              }>
              Newest
            </button>
            <button
              className="ProjectFilter"
              id="Oldest"
              onClick={[Function]}
              style={Object {}}>
              Oldest
            </button>
            <button
              className="ProjectFilter"
              id="A-Z"
              onClick={[Function]}
              style={Object {}}>
              A-Z
            </button>
            <button
              className="ProjectFilter"
              id="Z-A"
              onClick={[Function]}
              style={Object {}}>
              Z-A
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "height": "20px",
        "width": "100%",
      }
    } />
  <div
    className="ProjectList">
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#aed6f1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/ztaira14.github.io"
          onClick={[Function]}>
          ztaira14.github.io
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 19: A personal site built with React
      </p>
      <div>
        <span
          className="ProjectLanguage">
          JavaScript
        </span>
        <span
          className="ProjectDate">
          6/4/2017
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f5b7b1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/journal"
          onClick={[Function]}>
          journal
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        A public journal
      </p>
      <div>
        <span
          className="ProjectLanguage" />
        <span
          className="ProjectDate">
          5/29/2017
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#aed6f1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/ztaira14"
          onClick={[Function]}>
          ztaira14
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 19: The code behind the personal site I built with React
      </p>
      <div>
        <span
          className="ProjectLanguage">
          JavaScript
        </span>
        <span
          className="ProjectDate">
          5/15/2017
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#aed6f1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/pretty-reqs"
          onClick={[Function]}>
          pretty-reqs
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 18: A plain text graph editor, written with react. 
      </p>
      <div>
        <span
          className="ProjectLanguage">
          JavaScript
        </span>
        <span
          className="ProjectDate">
          4/5/2017
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#aed6f1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/abacus"
          onClick={[Function]}>
          abacus
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 17: An abacus desktop app using electron. 
      </p>
      <div>
        <span
          className="ProjectLanguage">
          JavaScript
        </span>
        <span
          className="ProjectDate">
          3/19/2017
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#aed6f1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/bsmsViz"
          onClick={[Function]}>
          bsmsViz
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 16: Visualizing the BS/MS program\'s depth courses in Northeastern\'s ECE Department
      </p>
      <div>
        <span
          className="ProjectLanguage">
          JavaScript
        </span>
        <span
          className="ProjectDate">
          2/3/2017
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#aed6f1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/gitbub"
          onClick={[Function]}>
          gitbub
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 15: Fun Github visualization using d3
      </p>
      <div>
        <span
          className="ProjectLanguage">
          JavaScript
        </span>
        <span
          className="ProjectDate">
          1/25/2017
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/httplescripts"
          onClick={[Function]}>
          httplescripts
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 14 w/ shoarora: Run applescripts remotely via HTTP requests
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          12/28/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/fb_creeper"
          onClick={[Function]}>
          fb_creeper
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 13: Python script to programmatically watch fb chat for you
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          11/26/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/hub_where"
          onClick={[Function]}>
          hub_where
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 11 and 12: Python scripts to gather and visualize data from Hubway\'s bike-sharing stations. 
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          10/2/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/evil_maid_suite"
          onClick={[Function]}>
          evil_maid_suite
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 10: Data visualization on the time I keylogged myself
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          9/23/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/souper_trawler"
          onClick={[Function]}>
          souper_trawler
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 9: A python script to scan a webpage and pull down relevant files.
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          9/10/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f5b7b1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/hackaweek"
          onClick={[Function]}>
          hackaweek
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 8.5: An example skeleton for all my hack-a-weeks. 
      </p>
      <div>
        <span
          className="ProjectLanguage" />
        <span
          className="ProjectDate">
          9/6/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f5b7b1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/zachs_resume"
          onClick={[Function]}>
          zachs_resume
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 8: My resume in LaTeX
      </p>
      <div>
        <span
          className="ProjectLanguage">
          TeX
        </span>
        <span
          className="ProjectDate">
          9/6/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f5b7b1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/whiz"
          onClick={[Function]}>
          whiz
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        A bookmark function for the terminal, because \'cd\' is too slow. 
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Shell
        </span>
        <span
          className="ProjectDate">
          8/30/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#a9dfbf",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/uku_tabs"
          onClick={[Function]}>
          uku_tabs
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 6 and 7: A terminal-based guitar/ukulele tab management tool. 
      </p>
      <div>
        <span
          className="ProjectLanguage">
          C++
        </span>
        <span
          className="ProjectDate">
          8/15/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f5b7b1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/vim_notes"
          onClick={[Function]}>
          vim_notes
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 5: A vim folding file for my notes. 
      </p>
      <div>
        <span
          className="ProjectLanguage">
          VimL
        </span>
        <span
          className="ProjectDate">
          8/1/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#a9dfbf",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/flappy_term"
          onClick={[Function]}>
          flappy_term
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 4: ASCII Flappy Bird clone using ncurses. 
      </p>
      <div>
        <span
          className="ProjectLanguage">
          C++
        </span>
        <span
          className="ProjectDate">
          7/30/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#a9dfbf",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/donkey.zach"
          onClick={[Function]}>
          donkey.zach
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 3: ASCII racing game inspired by the legendary DONKEY.BAS.
      </p>
      <div>
        <span
          className="ProjectLanguage">
          C++
        </span>
        <span
          className="ProjectDate">
          7/19/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#a9dfbf",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/python_flow"
          onClick={[Function]}>
          python_flow
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 2: A python flow visualizer in c++. Inspired by git log --graph.
      </p>
      <div>
        <span
          className="ProjectLanguage">
          C++
        </span>
        <span
          className="ProjectDate">
          7/15/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/aes_python"
          onClick={[Function]}>
          aes_python
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Hack-A-Week 1: An attempt at implementing the AES in python, for funsies.
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          7/3/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/absolutely_nothing"
          onClick={[Function]}>
          absolutely_nothing
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        The only code that never breaks. Ever. http://ztaira14.github.io/absolutely_nothing
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          4/10/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f5b7b1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/dotfiles"
          onClick={[Function]}>
          dotfiles
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Everything I need to to set up my personal coding environment. 
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Vim script
        </span>
        <span
          className="ProjectDate">
          3/29/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/RubiksGUI"
          onClick={[Function]}>
          RubiksGUI
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        A rubiks cube, implemented in tkinter. Operational via button input or key press. 
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          3/4/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/PythonGame"
          onClick={[Function]}>
          PythonGame
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        It\'s a bullet hell game in Python. It\'s pretty slow when too many objects are on the screen. Uses tkinter.
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          3/4/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/Email_Sender"
          onClick={[Function]}>
          Email_Sender
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Small email sender in Tkinter. Reads default email from a text file. Uses gmail. 
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          3/4/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/TwitterBot"
          onClick={[Function]}>
          TwitterBot
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        My first TwitterBot! :D
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          2/29/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#aed6f1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/HighChartsTest"
          onClick={[Function]}>
          HighChartsTest
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        A graph using highcharts
      </p>
      <div>
        <span
          className="ProjectLanguage">
          JavaScript
        </span>
        <span
          className="ProjectDate">
          2/26/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#aed6f1",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/JSTest"
          onClick={[Function]}>
          JSTest
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        This test bounces a ball.
      </p>
      <div>
        <span
          className="ProjectLanguage">
          JavaScript
        </span>
        <span
          className="ProjectDate">
          2/26/2016
        </span>
      </div>
    </div>
    <div
      className="Project"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#f9e79f",
        }
      }>
      <div>
        <a
          className="ProjectLink"
          href="https://github.com/ztaira14/myhistory"
          onClick={[Function]}>
          myhistory
        </a>
        <span
          className="favorite"
          onClick={[Function]}>
          ☆
        </span>
      </div>
      <p
        className="ProjectParagraph">
        Small wrapper to make using IDLE better. 
      </p>
      <div>
        <span
          className="ProjectLanguage">
          Python
        </span>
        <span
          className="ProjectDate">
          2/17/2016
        </span>
      </div>
    </div>
  </div>
</div>
`;
